contracts/modules/cover/Cover.sol

```diff
@@ -505,9 +555,28 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     CoverData storage cover =_coverData[coverId];
     CoverSegment memory segment = coverSegments(coverId, segmentId);
     PoolAllocation[] storage allocations = coverSegmentAllocations[coverId][segmentId];
+    ActiveCover storage _activeCover = activeCover[cover.coverAsset];
 
-    // TODO: implement using buckets
-    // totalActiveCoverInAsset[cover.coverAsset] -= burnAmount;
+    // Update expired buckets
+    {
+      uint lastUpdateId = _activeCover.lastBucketUpdateId;
+      uint currentBucketId = block.timestamp / BUCKET_SIZE;
+      _activeCover.lastBucketUpdateId = currentBucketId.toUint64();
+      uint expiredCoverAmount = expireActiveCoverBuckets(cover.coverAsset, lastUpdateId, currentBucketId);
+      // Reduce total active cover
+      uint burnSegmentBucketId = Math.divCeil((segment.start + segment.period), BUCKET_SIZE);
+      uint amountToBurn = (expiredCoverAmount + burnAmount);
+
+      // TODO: how to handle claims that come after cover segment has expired
+      _activeCover.totalActiveCoverInAsset = amountToBurn > _activeCover.totalActiveCoverInAsset
```

Comment:
This was a quick fix for the claims "double" expiry problem. Wasn't sure of a better way to handle this scenario.